src/images/calculate.py

```diff
@@ -72,43 +81,59 @@ def run():
     # this way images from more important articles will get a better scaling value.
-    images = {}
+    images_optional = {}
     preprocessed = preprocess.pages_selector.top_pages
     for file_position, (dir3, fname, _) in enumerate(preprocessed):
         # get the images that correspond to this file
+
         dskurls = page_images[(dir3, fname)]
 
-        # for each new image, save position of the file (article priority)
         for url in dskurls:
-            if url not in images:
-                images[url] = file_position
+            if required_enabled and image_is_required(url):
+                images_required.add(url)
+            # if it's an optional image, save position (priority) of the article it belongs
+            elif url not in images_optional:
+                images_optional[url] = file_position
 
-    # incorporate the dynamic images at the very top
+    # assign highest priority to dynamic images
     for url in dynamics:
-        images[url] = -1
-
-    total_images = len(images)
-    images = sorted(images.items(), key=operator.itemgetter(1))
+        images_optional[url] = -1
 
     # load image list to map disk paths to web addresses
     dskweb = {}
-    with codecs.open(config.LOG_IMAGENES, "r", encoding="utf-8") as fh:
-        for linea in fh:
-            dsk, web = linea.strip().split(config.SEPARADOR_COLUMNAS)
+    with open(config.LOG_IMAGENES, "rt", encoding="utf-8") as fh:
+        for line in fh:
+            dsk, web = line.strip().split(separator)
             dskweb[dsk] = web
 
-    logger.info("Calculating scales for %d images", total_images)
-    scaler = Scaler(total_images)
-    log_reduccion = codecs.open(config.LOG_REDUCCION, "w", encoding="utf8")
-    for i, (dskurl, _) in enumerate(images):
+    # all images saved to LOG_REDUCTION will be downloaded and scaled
+    log_reduction = open(config.LOG_REDUCCION, 'wt', encoding='utf8')
+
+    # save info of required images (won't be scaled)
+    for dskurl in images_required:
+        weburl = dskweb[dskurl]
+        info = '100', dskurl, weburl
+        log_reduction.write(separator.join(info) + '\n')
+
+    # sort optional images by assigned priority
+    images_optional = sorted(images_optional.items(), key=operator.itemgetter(1))
+
+    scaler = Scaler(len(images_optional))
+    for i, (dskurl, _) in enumerate(images_optional):
         scale = scaler(i)
         if scale == 0:
-            # done, no more images
-            log_reduccion.close()
-            return
+            # done, do not include more images
+            break
 
         weburl = dskweb[dskurl]
         info = (str(int(scale)), dskurl, weburl)
-        log_reduccion.write(config.SEPARADOR_COLUMNAS.join(info) + "\n")
+        log_reduction.write(separator.join(info) + "\n")
+
+    log_reduction.close()
+    logger.info("Images selected: required=%d optional=%d", len(images_required), i)
```

Comment:
Si vas a usar esta `i` afuera del for, ponele un mejor nombre (`pos`, `idx`, etc); quizás también estaría bueno un comentario tipo "esta variable refleja la cantidad de opcionales porque fue creciendo hasta el break o salir del for"

Reply:
Saqué el `enumerate` por que no estaba contemplado el caso de no tener imágenes opcionales.
Corregido en 6366ff5a20ad534cda38e3c41739e4ecd14f43a5